src/main/java/hudson/tasks/test/MatrixTestResult.java

```diff
@@ -52,7 +52,7 @@ public MatrixTestResult() {}
      */
     @Override
     protected String getChildName(AbstractTestResultAction tr) {
-        return ((MatrixBuild)tr.run).getProject().getName();
+        return tr.run.getParent().getName();
```

Comment:
This could be a `MatrixRun` not a `MatrixBuild`. `owner` is indeed deprecated and `run.getProject()` will not compile but this does.